internal/controller/controller.go

```diff
@@ -98,20 +106,15 @@ func StartControllers(ctx context.Context, config *rest.Config, logger logr.Logg
 
 	sink := newConfigSink(clientForConfigSink, kubeForConfigSink, logger, sinkChan)
 
-	if leaderElection {
-		logger.Info("Waiting to become the leader")
-		<-mgr.Elected()
-		logger.Info("Became the leader")
-	}
-
 	// Before starting the manager, initialize the config sink to sync all LLMBackend and LLMRoute objects in the cluster.
 	logger.Info("Initializing config sink")
+	ctx := context.Background()
 	if err = sink.init(ctx); err != nil {
 		return fmt.Errorf("failed to initialize config sink: %w", err)
 	}
 
 	logger.Info("Starting controller manager")
-	if err = mgr.Start(ctx); err != nil { // This blocks until the manager is stopped.
+	if err = mgr.Start(ctrl.SetupSignalHandler()); err != nil { // This blocks until the manager is stopped.
```

Comment:
i believe the signal handling is already setup from the main so no need to setup here again in another contex?

Reply:
I removed the signal handling from the main as it is the same as `ctrl.SetupSignalHandler`

Reply:
https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/manager/signals/signal.go#L30

Reply:
ah ok